WPFHospitalEditor/SearchResultDialog.xaml.cs

```diff
@@ -0,0 +1,219 @@
+            name.VerticalAlignment = VerticalAlignment.Center;
+        }
+
+        private void addAdvancedSearchButton()
+        {
+            Button advancedSearch = new Button();
+
+            adjustAdvancedSearchButtonProperties(advancedSearch);
+
+            advanceSearchButtons.Add(advancedSearch);
+
+            //advancedSearch.Click += Advanced_Search_Click(mapObject, i);
+
+            Grid.SetRow(advancedSearch, firstContentRowNumber);
+            Grid.SetColumn(advancedSearch, 4);
+
+            DynamicGrid.Children.Add(advancedSearch);
+        }
+
+        private static void adjustAdvancedSearchButtonProperties(Button advancedSearch)
+        {
+            advancedSearch.HorizontalAlignment = HorizontalAlignment.Center;
+            advancedSearch.VerticalAlignment = VerticalAlignment.Center;
+            advancedSearch.Content = "+";
+            advancedSearch.FontSize = 35;
+            advancedSearch.Foreground = Brushes.White;
+            advancedSearch.Background = Brushes.SkyBlue;
+            advancedSearch.FontWeight = FontWeights.UltraBold;
+            advancedSearch.Width = 35;
+            advancedSearch.Height = 35;
+            advancedSearch.HorizontalContentAlignment = HorizontalAlignment.Center;
+            advancedSearch.VerticalContentAlignment = VerticalAlignment.Top;
+            advancedSearch.BorderThickness = new Thickness(0);
+            advancedSearch.Padding = new Thickness(0, -10, 0, 0);
+        }
+
+        private void addSeparator()
+        {
+            Separator separator = new Separator();
+
+            separator.VerticalAlignment = VerticalAlignment.Bottom;
+
+            Grid.SetRow(separator, firstContentRowNumber);
+            Grid.SetColumn(separator, 1);
+            Grid.SetColumnSpan(separator, 4);
+
+            DynamicGrid.Children.Add(separator);
+        }
+
+        private int createBuildingRowData(MapObject mapObject)
+        {
+            addNameLabel(mapObject);
+
+            addAdvancedSearchButton();
+
+            addSeparator();
+
+            firstContentRowNumber++;
+
+            return firstContentRowNumber;
+        }
+
+        private void createOneRow(int height)
+        {
+            RowDefinition gridRow1 = new RowDefinition();
+            gridRow1.Height = new GridLength(height);
+            DynamicGrid.RowDefinitions.Add(gridRow1);
+        }
+
+        private void createRows()
+        {
+            for (int i = 0; i < HospitalMap.searchResult.Count; i++)
+            {
+                createOneRow(50);
+            }
+        }
+
+        private void Close_Click(object sender, RoutedEventArgs e)
+        {
+            HospitalMap.searchResult.Clear();
+            Close();
+        }
+
+        /*private void Advanced_Search_Click(MapObject mapObject, int i)
+        {
+            button.Click += new EventHandler(
+                (sender, e) => button_Click(sender, e, grid, i)
+            );
+        }*/
```

Comment:
same as before, commented code, and comments in general, should only be there if it's absolutely necessary

Reply:
explained, next feature